addons/website_slides/models/slide_channel.py

```diff
@@ -1002,14 +1002,13 @@ def _action_add_members(self, target_partners, member_status='joined', raise_on_
                 )
         return result_channel_partners
 
-    def _filter_add_members(self, target_partners, raise_on_access=False):
+    def _filter_add_members(self, raise_on_access=False):
         allowed = self.filtered(lambda channel: channel.enroll == 'public')
-        on_invite = self.filtered(lambda channel: channel.enroll == 'invite')
-        if on_invite:
-            if on_invite.has_access('write'):
-                allowed |= on_invite
-            elif raise_on_access:
-                raise AccessError(_('You are not allowed to add members to this course. Please contact the course responsible or an administrator.'))
+        if controlled_access := (self - allowed):
+            allowed += controlled_access._filtered_access('write')
+            if raise_on_access and allowed != self:
+                raise AccessError(_('You are not allowed to add members to this course. '
```

Comment:
Hello @flch-odoo 

As you know, we're trying to clean stalled tasks a bit.

Looking at this one, I think we could merge it, but maybe it would be nice to have a test with a "assertRaises" that makes sure this correctly fails when we want.

I checked and I did not see anything in the tests on "_action_add_members" with assertRaises so I don't think this is tested, maybe I did not look properly.

If you have an hour and want to do something else than documents, could be nice :disguised_face: 

Side-note: I don't think "CLN" is a proper commit tag, it's usually "REF" or "IMP", but I could be wrong.

Thanks & cheers!

Reply:
@flch-odoo ? 🥲

Reply:
@flch-odoo :pray: